packages/react-query/src/createTRPCReact.tsx

```diff
@@ -64,11 +64,15 @@ export interface ProcedureUseQuery<
       TData,
       TRPCClientErrorLike<TProcedure>
     >,
-  ): DefinedUseTRPCQueryResult<TData, TRPCClientErrorLike<TProcedure>>;
+  ): DefinedUseTRPCQueryResult<
+    // HACKYYYY
+    TData extends never[] ? inferTransformedProcedureOutput<TProcedure> : TData,
```

Comment:
these have nothing to do with fixing #4519, but a regression on #2996 after fixing #4519 where `placeholderData: []` infers the query result as `never[]` instead of the array type that the query returns:

![image](https://github.com/trpc/trpc/assets/51714798/2dd4fdb3-86b1-459d-931d-52cfcb6ad0a9)